### PR TITLE
[service.mailnotifiers] 0.2.10

### PR DESCRIPTION
**Pull request by senufo**
mail-notification is a service that allows you to see the arrival of new mail on the HOME page of xbmc

### CODE REVIEW

File: service.mailnotifiers/.gitignore [hunk 1]
```diff
@@ -0,0 +1,24 @@
+.gitignore
```
Comment (learningit) [inline service.mailnotifiers/.gitignore]: Please remove the gitignore from the PR.

File: service.mailnotifiers/addon.xml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  <extension point="xbmc.service"
+             library="notifier.py" start="login">
+  </extension>
+ <extension point="xbmc.addon.metadata">
+    <platform>all</platform>
+    <summary lang="en">Email Notifier</summary>
+    <summary lang="fr">Notification de l'arrivée des mails</summary>
+    <description lang="en">Notify new email[CR]Please note there is a bug with Aeon MQ5, messages superimposed. I have not yet found a solution[CR]For integration in the skin read README file
+    </description>
+    <description lang="fr">Notification de l'arrivée des mails[CR]Attention il y a un bug avec Aeon MQ5, les messages se superposent. je n'ai pas encore trouvé de solution[CR]Pour l'intégration dans un skin lisez le fichier LISEZMOI
```
Comment (learningit) [inline service.mailnotifiers/addon.xml]: Please add <source> and <forum> fields. Please see http://kodi.wiki/view/Addon.xml for more info.

File: service.mailnotifiers/notifier.py [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+"""
+kodi service for display number of mail on home screen.
+"""
+
+import xbmc, xbmcgui
+import xbmcaddon
+import poplib, imaplib
+import time
+
+__author__     = "Senufo"
```
Comment (learningit) [inline service.mailnotifiers/notifier.py]: in future updates, please remove the \__varname\__ style labels. these are reserved as internals in python.

File: service.mailnotifiers/notifier.py [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+            # Abort was requested while waiting. We should exit
+            break
+#        xbmc.log("hello addon! %s" % time.time(), level=xbmc.LOGDEBUG)
+
+# Verify if kodi work
+#while (not xbmc.abortRequested):
+    # Wait before get mails
+    #intervalle = int(float(Addon.getSetting('time')) * 60.0)
+        if start_time and (time.time() - start_time) < intervalle:
+            time.sleep(.5)
```
Comment (learningit) [inline service.mailnotifiers/notifier.py]: Do not use time.sleep, use xbmc.sleep() for values up to 2 secs, beyond that use monitor.waitForAbort().

**Comment by learningit** [comment]
I haven't tried running this, but am curious why the combination of creating a dialog window and then doing a xbmc.notification to display the msg cnt?

**Comment by senufo** [comment]
It is a choice in the settings. In addition to displaying the number of messges on the skin we can bring up a dialog box to the arrival of a new message.
I work to make the changes we need to change the version number or just a remake PR ?

**Comment by learningit** [comment]
You can add commits then squash them down into a single commit for the PR. See https://github.com/ginatrapani/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit for details.

**Comment by senufo** [comment]
I just make the changes and do the PR

File: service.mailnotifiers/notifier.py [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+            # elif re_added_control and not HomeNotVisible:
+            else:
+                # Try to get getcontrol if not exist make a new one
+                try:
+                    MsgBox = homeWin.getControl(MsgBoxId)
+                except:
+                    MsgBox = xbmcgui.ControlLabel(x, y, width, height, '', font, color)
+                # add control label and set default label
+                try:
+                    homeWin.addControl(MsgBox)
```
Comment (phil65) [inline service.mailnotifiers/notifier.py]: adding a control on-the-fly to home screen feels rather hacky.. not possible to just use a Notification?

**Comment by senufo** [comment]
I use this method for two years without problems. 
The advantage is that it helps to have the number of pending email even if the skin does not support this plugin. 
My main motivation for writing this service was able to see the number of unread email from KODI. The addition of the notification in the message box came after
I can put this optional feature

**Comment by phil65** [comment]
The foldername doesnt match the addon id. (service.mailnotifiers vs. service.mailnotifier)

**Comment by senufo** [comment]
I just correct this error

**Comment by phil65** [comment]
"Please note there is a bug with Aeon MQ5" <- could you explain that bug to me, please?

**Comment by senufo** [comment]
My bad.

This message concerns FRODO old version of the service. With the FRODO version of mailnotifiers and AEON MQ5 skin the information refresh not working.
For this version (ISENGARD) I tested with the confluence skin, amber and AEON MQ6 and everything works correctly.
I deleted this warning

**Comment by phil65** [comment]
Thx, should be fine for now then.
For next update perhaps you could fix a typo in your language file ("Fectching" instead of "Fetching")

**Pull request merged**